parkings/importers/parking_areas.py

```diff
@@ -0,0 +1,201 @@
+import sys
+import time
+from lxml import etree
+from owslib.wfs import WebFeatureService
+
+from django.contrib.gis.geos import Point, Polygon, MultiPolygon
+
+from parkings.models import ParkingArea
+
+
+class ParkingAreaImporter(object):
+
+    '''
+    Imports parking area data from kartta.hel.fi.
+
+    It should be possible to parallelize this a lot, for instance the wfs in
+    _download might be able to use maxfeatures and startindex while parsing and
+    saving can be multithreaded.
+    (see getfeature method in
+    https://github.com/geopython/OWSLib/blob/master/owslib/feature/wfs200.py)
+    However this only takes a few seconds currently and will probably be ran
+    at most daily so such optimization is probably premature at this point.
+    '''
+
+    def __init__(self, overwrite=False, verbosity=1):
+        # namespaces for lxml
+        self.ns = {
+            'wfs': 'http://www.opengis.net/wfs/2.0',
+            'avoindata': 'http://www.hel.fi/avoindata',
+            'gml': 'http://www.opengis.net/gml/3.2',
+        }
+        self.overwrite = overwrite
+        self.verbosity = verbosity
+
+        self.refusals = 0
+        self.overwrites = 0
+        self.created = 0
+
+    def import_areas(self):
+        start = time.time()
+
+        self._create_areas(self._parse_areas())
+
+        if self.verbosity >= 1:
+            if self.refusals != 0:
+                print(
+                    'Refused to overwrite data for %s areas because'
+                    ' --overwrite is not set.' % self.refusals
+                )
+            if self.overwrites != 0:
+                print('Overwrote data for %s parking areas.' % self.overwrites)
+            if self.created != 0:
+                print('Created %s new parking areas.' % self.created)
+            if (self.created + self.refusals + self.overwrites) == 0:
+                print('Nothing to do.')
+
+            print('Finished in %s seconds.' % round(time.time() - start, 2))
+
+    def _create_areas(self, area_dicts):
+        area_dicts_count = len(area_dicts)
+
+        for index, area_dict in enumerate(area_dicts):
+            if self.verbosity >= 1:
+                progress = 100 * ((index + 1) / area_dicts_count)
+                sys.stdout.write('\rCreating areas %d%%' % progress)
+
+            try:
+                parking_area = ParkingArea.objects.get(
+                    external_id=area_dict['external_id'],
+                )
+                self._update_parking_area(area_dict, parking_area)
+            except ParkingArea.DoesNotExist:
+                parking_area = self._create_parking_area(area_dict)
```

Comment:
I think using `get_or_create()` would be cleaner, something like:

```
created, parking_area = ParkingArea.objects.get_or_create(
   external_id=area_dict['external_id'],
   defaults={ ... }
)
if not created:
    self._update_parking_area( ... )
```

And then you need `save()` inside the update method, which actually makes more sense as you might skip saving if there is no overwrite.